### PR TITLE
ref(search): Remove ResultRenderer class in favor of functions

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is a bit more typical of how we write stuff.

### CODE REVIEW

File: static/app/components/searchSyntax/renderer.tsx [hunk 1]
```diff
@@ -105,12 +60,61 @@ type Props = {
...
 
-const FilterToken = styled('span')`
+const FilterToken = ({filter}: {filter: TokenResult<Token.Filter>}) => (
+  <Filter>
+    {filter.negated && <Negation>!</Negation>}
+    <KeyToken token={filter.key} negated={filter.negated} />
+    {filter.operator && <Operator>{filter.operator}</Operator>}
+    <Value>{renderToken(filter.value)}</Value>
+  </Filter>
+);
```
Comment (priscilawebdev) [inline static/app/components/searchSyntax/renderer.tsx:75]:
```suggestion
function FilterToken({filter}: {filter: TokenResult<Token.Filter>}) {
  return (
    <Filter>
      {filter.negated && <Negation>!</Negation>}
      <KeyToken token={filter.key} negated={filter.negated} />
      {filter.operator && <Operator>{filter.operator}</Operator>}
      <Value>{renderToken(filter.value)}</Value>
    </Filter>
  );
}
```

File: static/app/components/searchSyntax/renderer.tsx [hunk 2]
```diff
@@ -105,12 +60,61 @@ type Props = {
...
+  if (token.type === Token.KeyExplicitTag) {
+    value = (
+      <ExplicitKey prefix={token.prefix}>
+        {token.key.quoted ? `"${token.key.value}"` : token.key.value}
+      </ExplicitKey>
+    );
+  }
+
+  return <Key negated={!!negated}>{value}:</Key>;
+};
```
Comment (priscilawebdev) [inline static/app/components/searchSyntax/renderer.tsx:95]:
```suggestion
function KeyToken({
  token,
  negated,
}: {
  token: TokenResult<Token.KeySimple | Token.KeyAggregate | Token.KeyExplicitTag>;
  negated?: boolean;
}) {
  let value: React.ReactNode = token.text;

  if (token.type === Token.KeyExplicitTag) {
    value = (
      <ExplicitKey prefix={token.prefix}>
        {token.key.quoted ? `"${token.key.value}"` : token.key.value}
      </ExplicitKey>
    );
  }

  return <Key negated={!!negated}>{value}:</Key>;
}
```

File: static/app/components/searchSyntax/renderer.tsx [hunk 3]
```diff
@@ -105,12 +60,61 @@ type Props = {
...
+}: {
+  token: TokenResult<Token.ValueNumberList | Token.ValueTextList>;
+}) => (
+  <InList>
+    {token.items.map(({value, separator}) => [
+      <ListComma key="comma">{separator}</ListComma>,
+      renderToken(value),
+    ])}
+  </InList>
+);
```
Comment (priscilawebdev) [inline static/app/components/searchSyntax/renderer.tsx:108]:
```suggestion
function ListToken({
  token,
}: {
  token: TokenResult<Token.ValueNumberList | Token.ValueTextList>;
}) {
  return (
    <InList>
      {token.items.map(({value, separator}) => [
        <ListComma key="comma">{separator}</ListComma>,
        renderToken(value),
      ])}
    </InList>
  );
}
```

File: static/app/components/searchSyntax/renderer.tsx [hunk 4]
```diff
@@ -105,12 +60,61 @@ type Props = {
...
+
+const NumberToken = ({token}: {token: TokenResult<Token.ValueNumber>}) => (
+  <Fragment>
+    {token.value}
+    <Unit>{token.unit}</Unit>
+  </Fragment>
+);
```
Comment (priscilawebdev) [inline static/app/components/searchSyntax/renderer.tsx:115]:
```suggestion
function NumberToken({token}: {token: TokenResult<Token.ValueNumber>}) {
  return (
    <Fragment>
      {token.value}
      <Unit>{token.unit}</Unit>
    </Fragment>
  );
}
```

**Review by priscilawebdev — APPROVED** [review]
I left some suggestions but it looks good to me

File: static/app/components/searchSyntax/renderer.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply static/app/components/searchSyntax/renderer.tsx:75]: I like that there's less indention lol

**Pull request merged**